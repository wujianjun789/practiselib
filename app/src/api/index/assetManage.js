/**
 * Created by a on 2017/7/11.
 */
import {getHttpHeader, httpRequest} from '../../util/network'

export function getAssetManageModule(cb) {
    let headers = getHttpHeader();
    httpRequest('/', {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}

export function getDeviceProperty(cb) {
    let headers = getHttpHeader();
    httpRequest('/', {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}