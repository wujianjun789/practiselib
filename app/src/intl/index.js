import enLocaleData from 'react-intl/locale-data/en';
import zhLocaleData from 'react-intl/locale-data/zh';
import { addLocaleData } from 'react-intl';
import { getLocalStorage, setLocalStorage} from '../util/cache'
addLocaleData(enLocaleData);
addLocaleData(zhLocaleData);
const intl = {
    defaultLocale: 'zh',
}
function getIntl(lang) {
    intl.locale = lang.locale;
    intl.messages = lang.messages;
    return intl;
}
export function getDefaultIntl(cb) {
    let appLanguage = getLocalStorage("appLanguage");
    let newLanguage = appLanguage || 'zh';
    setLocalStorage("appLanguage", newLanguage);
    getTargetIntl(newLanguage, cb);
}

export function getIntlEn(cb) {
    if(typeof require.ensure === "function"){
        require.ensure(['./en'], () => {
            const en = require('./en').default;
            cb(getIntl(en));
        }, 'intl.en')
    }
}
export function getIntlZh(cb) {
    if(typeof require.ensure === "function"){
        require.ensure(['./zh'], () => {
            const zh = require('./zh').default;
            cb(getIntl(zh));
        }, 'intl.zh')
    }
}

export function getTargetIntl(locale, cb) {
    switch (locale) {
        case 'zh':
            return getIntlZh(cb);
        default:
            return getIntlEn(cb);
    }
}

