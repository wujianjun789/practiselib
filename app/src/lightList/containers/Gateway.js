/**
 * Created by Azrael on 2017/9/1.
 */
import '../../../public/styles/lightManage-list.less';
import React, { Component } from 'react';
import Content from '../../components/Content';
import SearchText from '../../components/SearchText';
import Select from '../../components/Select.1';
import TableWithHeader from '../components/TableWithHeader';
import TableTr from '../components/TableTr';
import Page from '../../components/Page';
import { getDomainList } from '../../api/domain';
import { getSearchAssets, getSearchCount, getDeviceStatusByModelAndId } from '../../api/asset';
import { getMomentDate, momentDateFormat } from '../../util/time';

export default class Gateway extends Component {
    constructor(props) {
        super(props);
        this.state = {
            page: {
                total: 0,
                current: 1,
                limit: 10
            },
            search: {
                value: '',
                placeholder: '请输入设备名称',

            },
            sidebarCollapse: false,
            currentDevice: null,
            deviceList: [],
            currentDomain: '',
            domainList: {
                titleField: 'name',
                valueField: 'name',
                options: []
            },
            currentControlMode: 'remote',
            controlModeList: {
                titleField: 'title',
                valueField: 'value',
                options: [
                    { title: '远程', value: 'remote' },
                    { title: '自动', value: 'auto' }
                ]
            }
        };

        this.model = 'gateway';

        this.columns = [
            { accessor: 'name', title: '设备名称' },
            { accessor: 'online', title: '通信状态' },
            { accessor: 'domain', title: '域' },
            { accessor: 'device', title: '设备状态' },
            { accessor: 'runningTime', title: '运行时间' },
            {
                accessor(data) {
                    return data.updateTime ? momentDateFormat(getMomentDate(data.updateTime, 'YYYY-MM-DDTHH:mm:ss Z'), 'YYYY/MM/DD HH:mm') : '';
                },
                title: '更新时间'
            }
        ];

        this.collapseHandler = this.collapseHandler.bind(this);
        this.onChange = this.onChange.bind(this);
        this.pageChange = this.pageChange.bind(this);
        this.searchChange = this.searchChange.bind(this);
        this.searchSubmit = this.searchSubmit.bind(this);
        this.tableClick = this.tableClick.bind(this);

        this.initData = this.initData.bind(this);
        this.updateDomainData = this.updateDomainData.bind(this);
        this.initDeviceData = this.initDeviceData.bind(this);
        this.updateDeviceData = this.updateDeviceData.bind(this);
        this.updatePageSize = this.updatePageSize.bind(this);
    }

    componentWillMount() {
        this.mounted = true;
        this.initData();
    }

    componentWillUnmount() {
        this.mounted = false;
    }

    initData() {
        getDomainList((data) => {
            this.mounted && this.updateDomainData(data, this.initDeviceData);
        });
    }

    updateDomainData(data, cb) {
        let currentDomain,
            options = data;
        if (data.length == 0) {
            currentDomain = null;
        } else {
            currentDomain = data[0];
        }
        this.setState({ domainList: { ...this.state.domainList, options }, currentDomain }, () => {
            cb && cb();
        });
    }

    initDeviceData(isSearch) {
        const { search: { value }, page, currentDomain } = this.state;
        if (isSearch) {
            page.current = 1;
            this.setState({ page: page });
        }

        const { limit, current } = this.state.page;
        const offset = limit * (current - 1);
        getSearchAssets(currentDomain ? currentDomain.id : null, this.model, value, offset, limit, this.mounted && this.updateDeviceData);
        getSearchCount(currentDomain ? currentDomain.id : null, this.model, value, this.mounted && this.updatePageSize);
    }

    updateDeviceData(data) {
        let currentDevice = data.length == 0 ? null : data[0];
        this.setState({ deviceList: data, currentDevice });
    }

    updatePageSize(data) {
        this.setState({ page: { ...this.state.page, total: data.count } })
    }

    onChange(e) {
        const { id, value } = e.target;
        switch (id) {
            case 'domain':
                let currentDomain = this.state.domainList.options[e.target.selectedIndex];
                this.setState({ currentDomain }, this.initDeviceData);
                break;
            case 'controlMode':
                this.setState({ currentControlMode: value });
                break;
        }
    }

    pageChange(page) {
        this.setState({ page: { ...this.state.page, current: page } }, this.initDeviceData);
    }

    searchChange(value) {
        this.setState({
            search: { ...this.state.search, value: value }
        })
    }

    searchSubmit() {
        this.initDeviceData(true);
    }

    collapseHandler() {
        this.setState({ sidebarCollapse: !this.state.sidebarCollapse });
    }

    tableClick(currentDevice) {
        this.setState({ currentDevice });
    }
    apply = () => {
        const { id } = this.state.currentDevice;
        const { currentControlMode } = this.state;
        console.log(id, currentControlMode)
        fetch(`/${id}/${currentControlMode}`)
            .then(res => {

            })
    }
    checkTime = () => {
        console.log('校验时间操作')
    }
    render() {
        const {
            page: { total, current, limit }, sidebarCollapse, currentDevice, deviceList,
            search: { value, placeholder }, currentDomain,
            domainList, currentControlMode, controlModeList
        } = this.state;
        const disabled = deviceList.length == 0 ? true : false;
        return <Content className={`list-lcc ${sidebarCollapse ? 'collapse' : ''}`}>
            <div className="content-left">
                <div className="heading">
                    <Select id='domain' titleField={domainList.titleField} valueField={domainList.valueField} options={domainList.options}
                        value={currentDomain == null ? '' : currentDomain[this.state.domainList.valueField]} onChange={this.onChange} />
                    <SearchText placeholder={placeholder} value={value} onChange={this.searchChange} submit={this.searchSubmit} />
                </div>
                <div className="table-container">
                    <TableWithHeader columns={this.columns}>
                        {
                            deviceList.map(item => <TableTr key={item.id} data={item} columns={this.columns} activeId={currentDevice.id}
                                rowClick={this.tableClick} willMountFuncs={[getDeviceStatusByModelAndId(this.model, item.id)]}
                            />)
                        }
                    </TableWithHeader>
                    <Page className={`page ${total == 0 ? "hidden" : ''}`} showSizeChanger pageSize={limit}
                        current={current} total={total} onChange={this.pageChange} />
                </div>
            </div>
            <div className={`container-fluid sidebar-info ${sidebarCollapse ? "sidebar-collapse" : ""}`}>
                <div className="row collapse-container" onClick={this.collapseHandler}>
                    <span className={sidebarCollapse ? "icon_horizontal" : "icon_verital"}></span>
                </div>
                <div className="panel panel-default panel-1">
                    <div className="panel-heading">
                        <span className="icon_select"></span>选中设备
                            </div>
                    <div className="panel-body">
                        <span title={currentDevice == null ? '' : currentDevice.name}>{currentDevice == null ? '' : currentDevice.name}</span>
                    </div>
                </div>
                <div className="panel panel-default panel-2">
                    <div className="panel-heading">
                        <span className="icon_touch"></span>设备操作
                            </div>
                    <div className="panel-body">
                        <div>
                            <span className="tit">控制模式：</span>
                            <Select id="controlMode" titleField={controlModeList.titleField} valueField={controlModeList.valueField}
                                options={controlModeList.options} value={currentControlMode} onChange={this.onChange} disabled={disabled} />
                            <button className="btn btn-primary" disabled={disabled} onClick={this.apply}>应用</button>
                        </div>
                        <div>
                            <span className="tit">校时：</span>
                            <span className="note">(点击以校准时间)</span>
                            <button className="btn btn-primary" disabled={disabled} onClick={this.checkTime}>校时</button>
                        </div>
                    </div>
                </div>
            </div>
        </Content>
    }
}

/**
 *  <Table columns={this.columns} keyField='id' data={deviceList} rowClick={this.tableClick}
                                activeId={currentDevice == null ? '' : currentDevice.id}/>
 */
