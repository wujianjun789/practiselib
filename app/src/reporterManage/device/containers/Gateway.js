

import React,{PureComponent} from 'react'

export default class Gateway extends PureComponent{
	render(){
		return(
			<div>this is gateway</div>
		)
	}
}