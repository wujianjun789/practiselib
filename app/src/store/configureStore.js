import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../common/reducer';
import thunk from 'redux-thunk';


export default function configureStore(initialState) {

    const composeEnhancers =
        process.env.NODE_ENV !== 'production' &&
            typeof window === 'object' &&
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
                // Specify here name, actionsBlacklist, actionsCreators and other options
            }) : compose;

    const enhancer = composeEnhancers(
        applyMiddleware(thunk),
        // other store enhancers if any
    );
    const store = createStore(rootReducer, initialState, enhancer);
    return store;

}
