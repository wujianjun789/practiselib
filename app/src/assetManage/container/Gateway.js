/**
 * Created by m on 2018/2/9.
 */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Content from '../../components/Content';

import { getModelData, getModelProps, getModelDefaultsValues, getModelDefaults, deleteModalTypes, addModalTypes }
  from '../../data/assetModels';
import { addModalTypesById } from '../../api/asset';
import Immutable from 'immutable';

import { injectIntl } from 'react-intl';
import { intlFormat } from '../../util/index';
import ConfirmPopup from '../../components/ConfirmPopup';
import TypeEditPopup from '../components/TypeEditPopup';
import { overlayerShow, overlayerHide } from '../../common/actions/overlayer';
import { getModelTypeByModel, updateModelTypeByModel } from '../../api/asset';
import { addNotify } from '../../common/actions/notifyPopup';
import NotifyPopup from '../../common/containers/NotifyPopup';

export class Gateway extends Component {
  constructor(props) {
    super(props);
    this.state = {
      model: 'ssgw',
      keyField: 'name',   //主键在数据表的Id属性中，即设备类型的id

      assetPropertyList: [   //设备属性列表
        // { id: '111', name: '三思LC', description: '三思单灯控制器', unit: '0000', accuracy: '0000' },
        // { id: '123', name: '华为LC_NBLot', description: '华为单灯控制器', unit: '0000', accuracy: '0000' },
      ],
      assetTypeList: [     //设备型号列表
        { name: '', description: '' }
        // { name: '华为单灯', description: '华为单灯控制器', power: '0000', life: '0000', manufacture: '0000' },
        // {name: '', detail: '', power: '', serviceLife: '', manufacture: '' },
      ],
    };


    this.assetTypeColumns = [  //ssslc设备型号表头定义types, name, description, power, life, manufacture
      { field: 'name', title: this.formatIntl('equipment.name')},
      { field: 'description', title: this.formatIntl('equipment.description') },
      { field: 'power', title: this.formatIntl('equipment.power') },
      { field: 'life', title: this.formatIntl('equipment.life') },
      { field: 'manufacture', title: this.formatIntl('equipment.manufacturer') },
    ];
    this.formatIntl = this.formatIntl.bind(this);
    this.rowDelete = this.rowDelete.bind(this);
    this.rowEdit = this.rowEdit.bind(this);
    this.rowAdd = this.rowAdd.bind(this);
    this.getModelType = this.getModelType.bind(this);
    this.updateModelType = this.updateModelType.bind(this);
  }

  componentDidMount() {
    let { model, assetPropertyList } = this.state;
    this.mounted = true;
    getModelData(() => {
      this.mounted && this.setState({ assetPropertyList: getModelProps(model) }, () => {
      });
    });
    this.getModelType();
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  formatIntl(formatId) {
    const { intl } = this.props;
    return intl ? intl.formatMessage({ id: formatId }) : null;
  }

  getModelType() {
    let { assetTypeList, model } = this.state;
    getModelTypeByModel(model, (data) => {
      let types = data[0].types;
      this.mounted && this.updateModelType(types);
    });
  }

  updateModelType(data) {
    let dataItem = { name: '', description: '' }
    data.push(dataItem);
    this.setState({ assetTypeList: data })
  }

  deleteModalTypeById(id) {
    const { assetTypeList, model } = this.state;
    let data = assetTypeList;
    let dataleft = _.remove(data, function (n) {
      return n.name === id;
    });
    data.pop();
    updateModelTypeByModel(model, data, (data) => {
      this.mounted && this.getModelType();
    })
  }

  rowDelete(id) { //id为型号名称
    const { data } = this.state;
    const { actions } = this.props;
    let curId = id; //当前要删除的型号的Id
    actions.overlayerShow(<ConfirmPopup iconClass="icon_popup_delete"
      tips={this.formatIntl('equipment.isDeleteDevice')} cancel={() => { actions.overlayerHide(); }}
      confirm={() => {
        this.deleteModalTypeById(curId);
        actions.overlayerHide();
      }} />);
  }

  rowEdit(id) { //id为选中的设备型号的name，
    let curId = id;
    const { assetTypeList, keyField, model } = this.state;
    const { actions } = this.props;
    let data = {};
    for (var key in assetTypeList) {
      if (curId === assetTypeList[key].name) {
        data = assetTypeList[key];
      }
    }

    actions.overlayerShow(<TypeEditPopup id="updateType"  title={this.formatIntl('equipment.addDevice')}
      data={data}
      idEdit={false}
      hasPower={true}
      hasLife={true}
      hasManufacture={true}
      onCancel={() => {
        actions.overlayerHide();
      }}
      onConfirm={(data) => {
        //data={type, power, serviceLife, manufacturer, detail,...others}
        //根据data中的参数更新设备型号，并将结果传入API实现数据的更改
        let name = data.name;
        let power = data.power;
        let description = data.description;
        let life = data.life;
        let manufacture = data.manufacture;
        let typeData = Object.assign({}, {
          name: name, power: power, description: description,
          life: life, manufacture: manufacture
        });
        //对更新后的数据做合并处理，名字是主键不能更改
        let list = assetTypeList;
        list.pop();
        let editData = list.map(item => {
          return item.name === typeData.name ? typeData : item
        })
        actions.overlayerHide();
        updateModelTypeByModel(model, editData, (data) => {
          this.mounted && this.getModelType();
        })
      }}></TypeEditPopup>);
  }

  rowAdd() { //添加设备状态，不传入Id，
    let curId = '';
    let { assetTypeList, model } = this.state;
    const { actions } = this.props;
    let data = {};
    data.name = '';
    data.description = '';
    data.power = '';
    data.life = '';
    data.manufacture = '';
    actions.overlayerShow(<TypeEditPopup id="updateType" title={this.formatIntl('equipment.addDevice')}
      addNotify={actions.addNotify}
      data={data}
      hasPower={true}
      hasLife={true}
      hasManufacture={true}
      onCancel={() => {
        actions.overlayerHide();
      }}
      onConfirm={(data) => {
        //data={type, power, serviceLife, manufacturer, detail, ...others}
        //根据data中的参数更新设备型号，并将结果传入API实现数据的更改
        let name = data.name;
        let power = data.power;
        let life = data.life;
        let manufacture = data.manufacture;
        let description = data.description;
        let typeData = Object.assign({}, {
          name: name, description: description
        });
        assetTypeList.pop()
        let dataAdd = assetTypeList;
        dataAdd.push(typeData);
        let resPram = ''
        updateModelTypeByModel(model, dataAdd, (data) => {
          this.mounted && this.getModelType();
          actions.overlayerHide();
        }, resPram, (msg) => {
          actions.addNotify(0, msg.message)
        })
      }}></TypeEditPopup>);
  }



  render() {
    const { data, assetPropertyList, assetTypeList, keyField } = this.state;
    let length = assetTypeList.length;
    return (
      <Content>
        <div className="row heading">
          <div className="type"><span></span>{this.formatIntl('asset.assetTypes')}</div>
          <table className="equipment">
            <thead>
              <tr>
                {
                  this.assetTypeColumns.map((column, index) => {
                    return <th key={index}>{column.title}</th>;
                  })
                }
                {<th>{this.formatIntl('equipment.edit')}</th>}
              </tr>
            </thead>
            <tbody>
              {
                assetTypeList.map((row, index) => {
                  return <tr key={index}>
                    {
                      this.assetTypeColumns.map((column, index) => {
                        return <td key={index}>{row[column.field]}</td>;
                      })
                    }
                    {index !== length - 1 ?
                      <td className="edit">
                        <a className="btn" role="presentation">
                          <span onClick={() => { this.rowEdit(row[keyField]); }} role="presentation"
                            className="icon_edit">{this.formatIntl('equipment.modify')}</span>
                        </a>
                        <a className="btn" role="presentation">
                          <span onClick={() => { this.rowDelete(row[keyField]); }} role="presentation"
                            className="icon_delete">{this.formatIntl('equipment.delete')}</span></a>
                      </td>
                      :
                      <td><span role="presentation" className="btn"
                        onClick={this.rowAdd}>+{this.formatIntl('equipment.add')}</span> </td>}
                  </tr>;
                })
              }
            </tbody>
          </table>
        </div>
      </Content>
    );
  }
}


function mapStateToProps(state) {
  return {
    sidebarNode: state.assetManage.get('sidebarNode'),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      addNotify: addNotify,
      overlayerShow: overlayerShow,
      overlayerHide: overlayerHide,
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(Gateway));