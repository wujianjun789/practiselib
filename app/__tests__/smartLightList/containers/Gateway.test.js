/**
 * Created by Azrael on 2017/9/27
 */
jest.mock('../../../src/util/network.js');
jest.mock('../../../src/api/asset.js');
jest.mock('../../../src/api/domain.js');
import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl-redux';
import configureStore from '../../../src/store/configureStore';

import Gateway from '../../../src/smartLightList/containers/Gateway';
import { getDefaultIntl } from '../../../src/intl/index'

getDefaultIntl();
const store = configureStore();
describe('<Gateway />', () => {
	it('default render', () => {
		const root = mount(<Provider store={store}>
			<IntlProvider>
				<Gateway />
			</IntlProvider>
		</Provider>);
		const cmp = root.find('Gateway');
		expect(!cmp.hasClass('collapse')).toBeTruthy();

		const select = cmp.find('#domain');
		// const domainList_state = cmp.state('domainList');
		// const currentDomain_state = cmp.state('currentDomain');
		// Object.keys(domainList_state).forEach(item => {
		// 	expect(select.prop(item)).toEqual(domainList_state[item]);
		// });
		// expect(select.prop('value')).toEqual(currentDomain_state == null ? '' : currentDomain_state[domainList_state.valueField]);

		// const searchText = cmp.find('SearchText');
		// const search_state = cmp.state('search');
		// Object.keys(search_state).forEach(item => {
		// 	expect(searchText.prop(item)).toBe(search_state[item]);
		// });

		// const tableWithHeader = cmp.find('TableWithHeader');
		// const inst = cmp.instance();
		// const columns = inst.columns;
		// expect(tableWithHeader.prop('columns')).toEqual(columns);

		// const tableTrs = cmp.find('TableTr');
		// const deviceList_state = cmp.state('deviceList');
		// const currentDevice_state = cmp.state('currentDevice');
		// deviceList_state.forEach((item, index) => {
		// 	let curTr = tableTrs.at(index);
		// 	expect(curTr.key()).toBe(item.id);
		// 	expect(curTr.prop('data')).toEqual(item);
		// 	expect(curTr.prop('columns')).toEqual(columns);
		// 	expect(curTr.prop('activeId')).toBe(currentDevice_state.id);
		// });

		// const page = cmp.find('Page');
		// const page_state = cmp.state('page');
		// expect(page.hasClass('hidden')).toBe(page_state.total == 0 ? true : false);
		// expect(page.prop('pageSize')).toBe(page_state.limit);
		// expect(page.prop('current')).toBe(page_state.current);
		// expect(page.prop('total')).toBe(page_state.total);

		const sidebarInfo = cmp.find('.sidebar-info');
		expect(!sidebarInfo.hasClass('sidebar-collapse')).toBeTruthy();

		// 测试icon className
		expect(sidebarInfo.find('.icon_verital').length).toBe(1);

		// const panel1 = sidebarInfo.find('.panel-1');
		// expect(panel1.find('.panel-heading').text()).toBe('选中设备');
		// expect(panel1.find('.panel-body span').text()).toBe(currentDevice_state == null ? '' : currentDevice_state.name);
		// expect(panel1.find('.panel-body span').props().title).toBe(currentDevice_state == null ? '' : currentDevice_state.name);

		// const panel2 = sidebarInfo.find('.panel-2');
		// expect(panel2.find('.panel-heading').text()).toBe('设备操作');
		// const panelBody = panel2.find('.panel-body');
		// expect(panelBody.children('div').length).toBe(2);

		// expect(panelBody.childAt(0).find('.tit').text()).toBe('控制模式：');
		// const controlModeList_state = cmp.state('controlModeList');
		// Object.keys(controlModeList_state).forEach(item => {
		// 	expect(panelBody.childAt(0).find('Select').prop(item)).toEqual(controlModeList_state[item]);
		// });
		// const currentControlMode_state = cmp.state('currentControlMode');
		// expect(panelBody.childAt(0).find('Select').prop('value')).toBe(currentControlMode_state);
		// expect(panelBody.childAt(0).find('Select').prop('disabled')).toBe(deviceList_state.length == 0 ? true : false);
		// expect(panelBody.childAt(0).find('button').text()).toBe('应用');
		// expect(panelBody.childAt(0).find('button').prop('disabled')).toBe(deviceList_state.length == 0 ? true : false);
        //
		// expect(panelBody.childAt(1).find('.tit').text()).toBe('校时：');
		// expect(panelBody.childAt(1).find('.note').text()).toBe('(点击以校准时间)');
		// expect(panelBody.childAt(1).find('button').text()).toBe('校时');
		// expect(panelBody.childAt(1).find('button').prop('disabled')).toBe(deviceList_state.length == 0 ? true : false);
	});

	it('simulate event', () => {
		const root = mount(<Provider store={store}>
			<IntlProvider>
				<Gateway />
			</IntlProvider>
		</Provider>);
		const cmp = root.find('Gateway');
		console.log('cmp:', cmp);
		let content = cmp.find('Content');
		expect(!content.hasClass('collapse')).toBeTruthy();
		content.find('.sidebar-info .collapse-container').simulate('click');
		content = cmp.find('Content');
		expect(content.hasClass('collapse')).toBeTruthy();
		expect(content.find('.sidebar-info').hasClass('sidebar-collapse')).toBeTruthy();
		expect(content.find('.sidebar-info .collapse-container span').hasClass('icon_horizontal')).toBeTruthy();
		content.find('.sidebar-info .collapse-container').simulate('click');
		content = cmp.find('Content');

		const domainList = {
			titleField: 'name',
			valueField: 'name',
			options: [{id: 1, name: '域一'},{id: 2, name: '域二'}]
		}
		root.setState({domainList});
		const _select = cmp.find('#domain');
		let event = {target: {id: 'domain', selectedIndex: 0}};
		_select.find('select').simulate('change', event);
		expect(_select.prop('value')).toBe(domainList.options[event.target.selectedIndex].name);

		const searchText = cmp.find('SearchText');
		event = {target: {value: '搜索'}};
		searchText.find('input').simulate('change', event);
		expect(searchText.prop('value')).toBe(event.target.value);

		const deviceList = [
			{id: 0, name: '设备1', online: true},
			{id: 1, name: '设备2', online: false}
		];
		const currentDevice = deviceList[0];
		root.setState({deviceList, currentDevice});
		const tableTrs = cmp.find('TableTr');
		console.log('tableTrs:', tableTrs);
		// tableTrs.at(1).find('tr').simulate('click');
		// expect(cmp.state('currentDevice')).toEqual(deviceList[1]);

		let controlMode = cmp.find('#controlMode');
		event = {target: { id: 'controlMode', value: 'off'}};
		controlMode.find('select').simulate('change', event);
		controlMode = cmp.find('#controlMode');
		expect(controlMode.prop('value')).toBe(event.target.value);
	});

	it('default render snapshot', () => {
		const cmp = renderer.create(<Provider store={store}>
			<IntlProvider>
				<Gateway />
			</IntlProvider>
		</Provider>);

		expect(cmp.toJSON()).toMatchSnapshot();
	})
})
