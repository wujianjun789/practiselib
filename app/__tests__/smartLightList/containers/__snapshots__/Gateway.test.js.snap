exports[`<Gateway /> default render snapshot 1`] = `
<div
  className="content list-lcc ">
  <div
    className="content-left">
    <div
      className="heading">
      <select
        className="form-control select "
        id="domain"
        onChange={[Function]}
        value="域一">
        <option
          value="域一">
          域一
        </option>
        <option
          value="域一">
          域一
        </option>
      </select>
      <div
        className="searchText "
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="请输入设备名称"
          type="search"
          value="" />
        <ul
          className="" />
        <span
          className="glyphicon glyphicon-search"
          onClick={[Function]} />
      </div>
    </div>
    <div
      className="table-container">
      <div
        className="table-responsive undefined">
        <table
          className="table table-hover">
          <thead>
            <tr>
              <th>
                设备名称
              </th>
              <th>
                通信状态
              </th>
              <th>
                设备状态
              </th>
              <th>
                调光模式
              </th>
              <th>
                更新时间
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      <ul
        className="rc-pagination page hidden"
        style={Object {}}
        unselectable="unselectable">
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-prev"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="上一页">
          <a />
        </li>
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="下一页">
          <a />
        </li>
        <div
          className="rc-pagination-options" />
      </ul>
    </div>
  </div>
  <div
    className="container-fluid sidebar-info ">
    <div
      className="row collapse-container"
      onClick={[Function]}>
      <span
        className="icon_verital" />
    </div>
    <div
      className="panel panel-default panel-1">
      <div
        className="panel-heading">
        <svg>
          <use
            height="200"
            transform="scale(0.08,0.08)"
            viewBox="0 0 20 20"
            width="200"
            x="0"
            xlinkHref="#icon_sys_select"
            y="0" />
        </svg>
        选中设备
      </div>
      <div
        className="panel-body">
        <span
          title="">
          
        </span>
      </div>
    </div>
    <div
      className="panel panel-default panel-2">
      <div
        className="panel-heading">
        <svg>
          <use
            height="200"
            transform="scale(0.088,0.086)"
            viewBox="0 0 20 20"
            width="200"
            x="0"
            xlinkHref="#icon_device_operate"
            y="0" />
        </svg>
        设备操作
      </div>
      <div
        className="panel-body">
        <div>
          <span
            className="tit">
            控制模式：
          </span>
          <select
            className="form-control select "
            disabled={true}
            id="controlMode"
            onChange={[Function]}
            value="remote">
            <option
              value="remote">
              远程
            </option>
            <option
              value="auto">
              自动
            </option>
          </select>
          <button
            className="btn btn-primary"
            disabled={true}
            id="controlMode_btn"
            onClick={[Function]}>
            应用
          </button>
        </div>
        <div>
          <span
            className="tit">
            校时：
          </span>
          <span
            className="note">
            (点击以校准时间)
          </span>
          <button
            className="btn btn-primary"
            disabled={true}
            id=" timing"
            onClick={[Function]}>
            校时
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
